### ISSUE TITLE
FieldtypeTextUnique outdated

### ISSUE THREAD

**Issue by tbba** [ryancramerdesign/FieldtypeTextUnique#3]
Hi, on the late PW betas, FieldtypeTextUnique starts throwing SQL errors here.
Comparing the code with FieldtypeText, the code there looks kind of outdated.

Details, if that matters:
I used it on a fresh PW install where I tried to set a new field and got this:
SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 1000 bytes (checkFieldTable)
In "fields" the field was created as a new entry, but there was no associated field table installed with it.
(On existing PW installs, the field still works, just with the fresh PW beta install the SQL I saw this, where using normal FieldtypeText work fine for the new field.)

And a long-standing wish for this fieldtype:
When duplicating a page using a FieldtypeTextUnique field, instead of throwing a "duplicate found" error maybe just leave the field blank on the newly created page copy?

Thank you for all your work in ProcessWire. It turned out to be one of the best ideas in my professional live starting using it. We do really crazy stuff with it, far beyond normal websites.

**Comment by tbba** [comment]
btw. I tink I ran into the Maria database problem mentioned in the other open issue.

**Comment by tbba** [comment]
Hi, thank you very much for updating this. It helps a lot.

It works very well. 

2 small observations.

1.
I changed the needed field from text to textUnique (option checked "leave settings"):
Fatal error:
**QLSTATE[42S01]: Base table or view already exists: 1050 Table 'field_page_key_pwtmp' already exists**
Refreshing the site removed the fatal error but the name had changed from "page_key" to "page_key_PWTMP".
Of course, I could rename the field back afterwards. No further error.

2.
In the module settings I see no link to this GitHub repo. (That’s why I thought a few days ago this module became part of the PW core).

**Comment by tbba** [comment]
There is another error now in the front end:
Field sortierung is not saveable because it is in a specific context 

![Bildschirmfoto 17](https://user-images.githubusercontent.com/652879/118604500-8a2b7700-b7b5-11eb-8695-26bb20a816bc.jpg)

**Comment by tbba** [comment]
The last error went away, when re-saving existing textUnique fields (like the field "sortierung" in the error example above).

**Comment by ryancramerdesign** [comment]
@tbba PW won't save Field objects that are within the context of a particular template. I'm guessing the module did that on its own, so likely needs a check to make sure the field isn't a contextual one. I'll add that, thanks. 

For changing between text and textUnique, I can't seem to duplicate the issue here yet. I've been changing back and forth. What's your PW core version?

I don't usually put links to the GitHub repo in my modules, but can if it helps? The way to identify core modules from non-core is the location fo the module files. Core modules are in /wire/modules/ while non-core modules are in /site/modules/.

**Comment by tbba** [comment]
I used PW with the latest DEV 3.0.178

>  I've been changing back and forth 

That wasn't the issue. It only happened when switching for the first time from text to unique. After a saving the problem did not come back. At some stage the pwtmp is added to the field. At that state the problem shows. After removing that from the field name manually and saving, the problem never came back. I do not think many people will ever notice that problem, so I just wanted to mention it.

> I don't usually put links to the GitHub repo in my modules, but can if it helps

With that, you help direct the users to the right spot to leave comments and bug reports i.m.h.o.
Like people are used to from 3rd party modules already. At least I got confused at first, and since you were the author, I thought that the core-issues repo was the right spot for my note.

**Comment by ryancramerdesign** [comment]
@tbba I'm not sure about the pwtmp issue you mentioned, as that sounds like core rather than this module. So far I can't duplicate, but I am thinking potentially it was a side effect of the issue in your screenshot above. I have pushed a fix for that issue, so hopefully it prevents either issue from occurring again. I have also added the GitHub URL to the getModuleInfo() return value, I think you are right it's good to have that be easily accessible, especially for issue reports.
